### DISCUSSION TITLE
Update/Installstion broken? (composer)

### DISCUSSION THREAD

**Discussion by JanFeld** [firefly-iii/firefly-iii#5818] [Firefly III] [answered]
Hi,

i tried updating from 5.6.8 to 5.6.14 the same way i’ve done jt many times.

Now i’m getting the following error and can’t update:

```
Your requirements could not be resolved to an installable set of packages.

Problem 1
    - Installation request for doctrine/dbal 3.3.1 -> satisfiable by doctrine/dbal[3.3.1].
    - doctrine/dbal 3.3.1 requires composer-runtime-api ^2 -> no matching package found.

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://getcomposer.org/doc/04-schema.md#minimum-stability> for more details.
 - It's a private package and you forgot to add a custom repository to find it 
 ```

Is there something broken or did i miss something regarding the last releases?

A side question:
I’m thinking about migrating FF3 to docker. Is there an easy way to migrate the database (mariadb) to the docker compose installation? I tried it once a long time ago but couldn’t manage to restore the data. I don’t want to lose the last three years of data i’ve put into FF3.. ;-)

Thanks for your answers!

Cheers,
Jan

**Comment by JC5** [comment, marked as answer]
1. Perhaps your composer is too old, try upgrading it. Otherwise, I can't think of anything.
2. If you can export the database to an SQL file, you can import it again. You can do this with phpMyAdmin or with a tool like Querious or Sequel Pro (sorry, I only know Mac tools). Both tools can connect to either a docker container (if you forward the database port) or to the original DB.

In regards to the migration: allow Firefly III to start up and run in the background (with a fresh database). Then, open your database app and delete all tables and import the set from your previous installation. Restart the Firefly III container to finish the process.

**Comment by JanFeld** [comment]
I couldn’t fix the composer issue yet but managed to get the DB data into the Docker Database/App.

I’ve added an phpMyAdmin container to the compose-file and increased the maximum upload data size to import the DB.
I’ve got it running using an nginx reverse proxy.

I’m struggling with using the SSL connection which is set up similarly to an instance of wekan (also docker) which is configured the same way.
I have some trouble with loading data like the diagrams/dashboard and some pages. They seem like there are some connection problems and some pages look like they are missing all UI elements except the HTML stuff. But that’s another story.

Nevertheless thanks for your hints and answer.

Cheers,
Jan

**Reply by JC5** [reply]
I think composer has a verbal option, `-vvv`. That could work to see what happens and why it can't install dbal.

**Reply by JanFeld** [reply]
Thanks for your answer.
I've successfully migrated to Docker.
I think there's no newer version of composer available for my current distro. Maybe it's time to upgrade..
